src/main/java/io/cryostat/core/reports/InterruptibleReportGenerator.java

```diff
@@ -104,17 +106,28 @@ public InterruptibleReportGenerator(
     }
 
     public Future<ReportResult> generateReportInterruptibly(InputStream recording) {
+        return generateReportInterruptibly(recording, rule -> true);
+    }
+
+    public Future<ReportResult> generateReportInterruptibly(
+            InputStream recording, Predicate<IRule> predicate) {
         return qThread.submit(
                 () -> {
                     // this is generally a re-implementation of JMC JfrHtmlRulesReport#createReport,
                     // but calling our cancellable evalute() method rather than the
                     // RulesToolkit.evaluateParallel as explained further down.
+                    Objects.requireNonNull(predicate);
+                    Objects.requireNonNull(recording);
```

Comment:
It would still be better to move these outside of the `qThread`, I think. The failure will occur a little sooner, and you'll probably get a cleaner stack trace out of it too.

Reply:
Oh, yeah I don't know why but I was thinking that this was going on a new thread so that the extra null checks wouldnt affect the main thread, but that doesnt make sense now that I see it.